Frameworks/CoreGraphics/CGPath.mm

```diff
@@ -192,6 +149,24 @@ void _CGPathGetBoundingBoxInternal(CGPathRef pathref, CGRect* rectOut) {
     pathref->_getBoundingBox(rectOut);
 }
 
+void _CGPathAddElement(
+    CGPathRef path, CGPathElementType type, CGPoint p0 = CGPointZero, CGPoint p1 = CGPointZero, CGPoint p2 = CGPointZero) {
+    if (path->_count + 1 >= path->_max) {
+        path->_max += 32;
+        path->_elements = (CGPathElementInternal*)IwRealloc(path->_elements, path->_max * sizeof(CGPathElementInternal));
+    }
+    CGPathElementInternal* element = &path->_elements[path->_count];
+    // The new elements needs its points array pointer updated
+    // because it was created with alloc
+    element->updatePointsArrayPointer();
```

Comment:
nit: from a OO standpoint, i would call this function init() (caller doesn't need to know internal details of what this function does).  There is alternately, no need for this function, you can call placement new on the memory location like this: CGPathElementInternal* element = new (&path->_elements[path->_count]) CGPathElementInternal();